### PR TITLE
feature: Use static binary instead of docker image

### PR DESCRIPTION
**Pull request by lolgab**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -42,13 +42,18 @@ outputs:
...
+        download_url=$(curl https://api.github.com/repos/codacy/git-version/releases/latest | jq -r .assets[0].browser_download_url)
+        curl -Ls $download_url > bin/git-version
     - id: previous-version
       shell: bash
       run: |
         set -e
 
-        export PREVIOUS_VERSION=$(docker run -v $(pwd):/repo codacy/git-version:${{ inputs.tool-version }} \
-          /bin/git-version \
+        export PREVIOUS_VERSION=$(bin/git-version \
```
Comment (h314to) [inline action.yml:56]: you may need to run `git config --global --add safe.directory /repo` before running git version if you run into that error caused folder ownnership

**Review by h314to — APPROVED** [review]

**Review by afsmeira — COMMENTED** [review]
With this change `tool-version` should no longer be necessary and I think we can delete it, making this a breaking change.

But why not take it one step further and download the binary version matching the action version?

File: action.yml (hunk 1 above)
Comment (lolgab) [reply action.yml:56]: We're using Github Action git which should have all the permissions, no?

Comment (lolgab) [reply action.yml:56]: But I can test it on pulse to see what happens.

**Comment by lolgab** [comment]
> With this change `tool-version` should no longer be necessary and I think we can delete it, making this a breaking change.

Actually, this wasn't ready for review yet since I wanted to try the approach first and then port the `tool-version` so it uses that if defined.

> But why not take it one step further and download the binary version matching the action version?

It is what I actually wanted!
The thing is... I don't know how to do that, and if it's possible at all to reference the version of the action itself from the action.

**Comment by lolgab** [comment]
Tested in pulse [here](https://github.com/codacy/pulse/actions/runs/3524960814/jobs/5911058714)

**Review by lventura-codacy — APPROVED** [review]

**Pull request merged**